src/viam/sdk/tests/test_robot.cpp

```diff
@@ -1,323 +1,212 @@
+    BOOST_CHECK(Registry::lookup_resource(Generic::subtype(), generic_model));
+    BOOST_CHECK(Registry::lookup_resource(Motor::subtype(), motor_model));
 }
 
-viam::component::arm::v1::Status default_status() {
-    viam::common::v1::Pose pose;
-    pose.set_x(1);
-    pose.set_y(2);
-    pose.set_z(3);
-    pose.set_o_x(4);
-    pose.set_o_y(5);
-    pose.set_o_z(6);
-    pose.set_theta(20);
-
-    viam::component::arm::v1::JointPositions jp;
-    for (auto i = 0; i < 6; ++i) {
-        jp.add_values(0);
+template <typename T>
+// Sorts our vecs and converts to strings for consistent comparisons
+std::vector<std::string> vec_to_string_util(std::vector<T>& vec) {
+    std::vector<std::string> ret;
+    for (auto& v : vec) {
+        ret.push_back(v.SerializeAsString());
     }
+    std::sort(ret.begin(), ret.end());
+    return ret;
+}
 
-    viam::component::arm::v1::Status status;
-    *status.mutable_end_position() = pose;
-    *status.mutable_joint_positions() = jp;
-    status.set_is_moving(false);
+BOOST_AUTO_TEST_CASE(test_resource_names) {
+    server_to_client_pipeline(
+        [](std::shared_ptr<RobotClient> client, MockRobotService& service) -> void {
+            std::vector<ResourceName>* resource_names = client->resource_names();
+            auto names = vec_to_string_util(*resource_names);
+            auto mocks = mock_resource_names_response();
 
-    return status;
+            auto mock_resp = vec_to_string_util(mocks);
+            BOOST_CHECK(names == mock_resp);
+        });
 }
 
-::grpc::Status TestService::FrameSystemConfig(
-    ::grpc::ServerContext* context,
-    const ::viam::robot::v1::FrameSystemConfigRequest* request,
-    ::viam::robot::v1::FrameSystemConfigResponse* response) {
-    *response->mutable_frame_system_configs() = config_response();
-    return ::grpc::Status();
+BOOST_AUTO_TEST_CASE(test_get_status) {
+    server_to_client_pipeline(
+        [](std::shared_ptr<RobotClient> client, MockRobotService& service) -> void {
+            auto mock_statuses = mock_status_response();
+
+            // get all resource statuses
+            auto statuses = client->get_status();
+
+            auto status_strs = vec_to_string_util(statuses);
+            auto mock_strs = vec_to_string_util(mock_statuses);
+
+            // ensure we get statuses for all resources, and that they are as expected.
+            BOOST_CHECK(statuses.size() == 3);
+            BOOST_CHECK(vec_to_string_util(statuses) == vec_to_string_util(mock_statuses));
+
+            // get only a subset of status responses
+            auto names = mock_resource_names_response();
+            std::vector<ResourceName> some_names{names[0], names[1]};
+            auto some_statuses = client->get_status(some_names);
+            auto some_status_strs = vec_to_string_util(some_statuses);
+
+            // ensure that we only get two of the three existing statuses
+            BOOST_CHECK(some_status_strs.size() == 2);
```

Comment:
[nit] BOOST_CHECK_EQUAL will also print the value of `some_status_strs.size()` if the comparison fails